interpreter/cling/lib/Interpreter/IncrementalParser.cpp

```diff
@@ -106,8 +106,23 @@ namespace {
     return false;
   }
 
+  /// \brief Overrides the current DiagnosticConsumer to supress many warnings
+  /// issued as a result of incremental compilation (see `HandleDiagnostic()`).
+  ///
+  /// Diagnostics passing the filter are, by default, forwarded to the previous
+  /// DiagnosticConsumer instance.  A different consumer may be specified via
+  /// `setTargetConsumer()`.
+  /// In that case, given that internal state might be updated as part of
+  /// `{Begin,End}SourceFile` (e.g. in TextDiagnosticPrinter), calls to such
+  /// functions will be forwarded to both, the user-specified and the original
+  /// consumer; however `HandleDiagnostic()` calls shall only be seen by the
+  /// former.
+  ///
+  /// On destruction, the original (i.e. overridden consumer) is restored.
```

Comment:
Maybe the comment is a bit misleading, I don't see this happening in the destructor `~FilteringDiagConsumer`; it only appears to free the target `DiagnosticConsumer` in case it is owed.

Reply:
That happens in `~ReplaceDiagnostics`, i.e. a base class destructor.